ipa-core/src/helpers/transport/query/mod.rs

```diff
@@ -260,43 +259,45 @@ impl FromStr for DPParams {
     }
 }
 
-// TODO: naming: s/TestingWithNoDp/no-dp
-impl Display for DPParams {
+impl Display for DpParams {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
-            DPParams::TestingWithNoDP => write!(f, "TestingWithNoDP"),
-            DPParams::WithDP(eps) => write!(f, "WithDP={eps}"),
+            DpParams::NoDp => write!(f, "NoDp"),
+            DpParams::WithDp(eps) => write!(f, "WithDp={eps}"),
         }
     }
 }
 
-impl Serialize for DPParams {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where S: Serializer {
+impl Serialize for DpParams {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
         serializer.serialize_str(&self.to_string())
```

Comment:
Why have this as a string and not a float?  NoDp can be Infinity.

Reply:
we'd been having trouble with serialization and @akoshelev helped come up with this to fix it.  But I think he also thought it was something we should revisit and try some other approaches.

Reply:
In fact @akoshelev  created an issue related to this. https://github.com/private-attribution/ipa/issues/1149#issue-2360310674  

I'd hope we can get it working with the current approach for this PR and then align on what we want to do longer term.

Reply:
Works for me.

Reply:
yea, this is ugly. Ben is right - it all comes down to our earlier decision to pass query params through the query string. The serialization options are extremely limited there: `serde_urlencoded` supports just plain structs and simple enums, failing at runtime for anything more complicated than that.

Once we fix #1149, we can eliminate this custom serializer